### PR TITLE
upvote feature

### PR DESCRIPTION
**Pull request by ExplorerAadi**
Upvote feature

### CODE REVIEW

File: commands/upvote.js [hunk 1]
```diff
@@ -1,39 +1,71 @@
...
-                    .fetch(process.env.CM_BOT_CHANNEL || botChannel)
-                    .then((channel) => {
-                        channel.send(
-                            `<@!${message.author.id}>, the user you mentioned is not registered in the DB. Your DM is not accessible. Please enable it **User settings > Privacy & safety > Allow messages from server members**`
-                        );
-                    });
-            }
-        }
+    const member = await Member.findOne({
+      discordId: message.mentions.users.first().id,
```
Comment (beingcodr) [inline commands/upvote.js]: Change `message.mentions.users.first().id` to `mentionedUser.id`

File: commands/upvote.js [hunk 2]
```diff
@@ -1,39 +1,71 @@
...
+    });
+    if (!member) {
+      try {
```
Comment (beingcodr) [inline commands/upvote.js]: Change the entire `try catch` to 

```
messageErrorAsync(
    message,
    `<@!${mentionedMember.id}> is not registered in the database`,
    `<@!${message.author.id}>, the user you mentioned is not registered in the DB`
)
```

File: commands/upvote.js [hunk 3]
```diff
@@ -1,39 +1,71 @@
...
-            default:
-                break;
-        }
-    },
+    const { codeDoubt, codeError, verbalDoubt, sharedResource } = pointInc;
+
+    pointInc.contribution += 1;
+    totalPointsCal += codeDoubt + codeError + verbalDoubt + sharedResource;
+
+    if (!args.length > 1) {
```
Comment (beingcodr) [inline commands/upvote.js]: The condition should be `if(args.length === 1)`

File: commands/upvote.js [hunk 4]
```diff
@@ -1,39 +1,71 @@
...
+      if (args === "codeError") {
```
Comment (beingcodr) [inline commands/upvote.js]: Make theses `if`'s to `switch` case and add a `default` as **Invalid argument**

File: commands/upvote.js [hunk 5]
```diff
@@ -1,39 +1,71 @@
...
+        verbalDoubt += 5;
+      }
+      if (args === "codeDoubt") {
+        codeDoubt += 10;
+      }
+      if (args === "sharedResource") {
+        sharedResource += 5;
+      }
+    } else {
+      try {
```
Comment (beingcodr) [inline commands/upvote.js]: Change the entire `try catch` to

```
messageErrorAsync(
    message,
    `Please upvote for only one type of help at a time.`,
    `<@!${message.author.id}>, please upvote for only one type of help at a time`
)
```

File: commands/upvote.js [hunk 6]
```diff
@@ -1,39 +1,71 @@
...
+        await message.author.send(
+          "Please upvote for only one type of help at a time."
+        );
+      } catch (error) {
+        console.log(error);
+      }
+    }
+
+    Member.update(
```
Comment (beingcodr) [inline commands/upvote.js]: Could you refactor this code to **Mongoose** syntax like `member.save()` and all. You can refer the file `commands/addRole.js` after you pull the **dev** branch

File: commands/upvote.js [hunk 7]
```diff
@@ -1,39 +1,71 @@
...
+              `<@!${message.author.id}>, the user you mentioned is not registered in the DB. Your DM is not accessible. Please enable it **User settings > Privacy & safety > Allow messages from server members**`
+            );
+          });
+      }
+    }
 
-        switch (args.slice(1)) {
-            case value:
-                break;
+    let pointInc = member[points],
```
Comment (beingcodr) [inline commands/upvote.js]: Add `args = args.slice(1).join('')` above this code, to get rid of the **@mentioned user** from the **args**

**Review by beingcodr — CHANGES_REQUESTED** [review]

**Comment by imrhlrvndrn** [comment]
@Aadi-27 You also have to `git pull` **dev** branch to get all the updated code. The followings steps have to be followed to have the updated code without any complications:

1. `git stash` on the local repository, to stash all the modified files
1. `git pull upstream dev` to pull the updated code
1. It is possible that the **pull** will result in **2 merge conflicts** in the `upvote.js` file. To resolve that:
    - Navigate to the `upvote.js` file in Visual Code
    - Click on the **Accept Incoming changes** to update the code with the latest code and preserve the new code you wrote below
    - Clicking **Accept Current changes** would result in keeping the old code, which you'll have to update yourself

**Review by beingcodr — APPROVED** [review]

**Pull request merged**